### ISSUE TITLE
etcd health flapping on HA Rancher cluster (RKE)

### ISSUE THREAD

**Issue by StoneCut** [rancher/rke#1938]
We run a Rancher HA cluster built with RKE. Load balancing is done by a Citrix NetScaler.
I see almost constant flapping "**Alert: Component etcd-1 is unhealthy**." (or **etcd-0** or **etcd-2** and sometimes even all of them). I have updated the cluster (effectively reinitializing everything) and still have this issue.

Please help troubleshoot.

**RKE version:**
1.0.0
**Docker version: (`docker version`,`docker info` preferred)**
19.03.6
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
Ubuntu 18.04.4 - 5.3.0-28-generic
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VMware ESX
**cluster.yml file:**
**Steps to Reproduce:**

**Results:**

**Comment by superseb** [comment]
Please share the specs of the machine (CPU/memory/disk type and speed) and the logs from the etcd containers (`docker logs etcd`). Using `kubectl get cs -o yaml` will get you a bit more info on the error (if you run it while it happens).

**Comment by StoneCut** [comment]
> Please share the specs of the machine (CPU/memory/disk type and speed) and the logs from the etcd containers (`docker logs etcd`). Using `kubectl get cs -o yaml` will get you a bit more info on the error (if you run it while it happens).

Thank you for your reply! Due to your help I've been able to determine that it's appa
[etcd.log](https://github.com/rancher/rke/files/4286277/etcd.log)
rently really one node acting up. The etcd container keeps restarting and the other nodes give bad health when trying to query this one node. I've attached the (anonymized) log file of the etcd container on the problematic node.

The specs of all three nodes are:
8 GB RAM
2 vCPU
Disk type and speed is hard to determine since this is an ESX HA-Cluster. I don't see high "wa" (IO Wait?) values in top, though.
